include/seqan3/core/metafunction/iterator.hpp

```diff
@@ -97,7 +98,7 @@ template <std::InputIterator it_t>
 struct rvalue_reference<it_t>
 {
     //!\brief Return the member type as return type.
-    using type = ranges::rvalue_reference_t<it_t>;
+    using type = decltype(ranges::iter_move(std::declval<it_t &>()));
```

Comment:
This still depends on the ranges library? Why is this any better?

Reply:
`rvalue_reference_t` won't be part of C++20, `iter_move` will be.

Reply:
and pending the changes by @joshuak94 it would become `std::ranges::iter_move`